### PR TITLE
Enable nullability in ChildAccessibleObject

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in ChildAccessibleObject

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6287)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ChildAccessibleObject.cs [hunk 1]
```diff
@@ -14,7 +12,7 @@ public class ChildAccessibleObject : AccessibleObject
         {
             private readonly ComboBox _owner;
 
-            public ChildAccessibleObject(ComboBox owner, IntPtr handle)
+            public ChildAccessibleObject(ComboBox? owner, IntPtr handle)
             {
                 Debug.Assert(owner?.IsHandleCreated is true, "ComboBox's handle hasn't been created");
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ChildAccessibleObject.cs:18]: This doesn't look correct - if `[in] owner` is nullable, then the backing field must be nullable as well.

**Review by RussKie — CHANGES_REQUESTED** [review]
@vladimir-krestov @SergeySmirnov-Akvelon please review whether `[in] owner` can be `null` at all. If not - then we must throw ANE like we do in other places.

File: src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ChildAccessibleObject.cs [hunk 2]
```diff
@@ -14,7 +12,7 @@ public class ChildAccessibleObject : AccessibleObject
         {
             private readonly ComboBox _owner;
 
-            public ChildAccessibleObject(ComboBox owner, IntPtr handle)
+            public ChildAccessibleObject(ComboBox? owner, IntPtr handle)
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ChildAccessibleObject.cs:15]: The `owner` **can't** be `null` in the current implementation.
Check the constructor links. It's used for `ComboBoxChildEditUiaProvider` and `ComboBoxChildListUiaProvider` classes. Their constructors are used once, and `owner` argument is a ComboBox (check links, there are `this` is used).
Moreover, `ComboBoxChildEditUiaProvider` and `ComboBoxChildListUiaProvider` use `_owningComboBox` and `_owner` fields without nullablity checks. So we are sure, that they can't be null. And in the end we should mark `owner` as non-nullable argument.
```suggestion
            public ChildAccessibleObject(ComboBox owner, IntPtr handle)
```

@gpetrou, please:
- remove `?` from this file and make `owner/_owner` as non-nullable. 
- remove `PublicAPI.Shipped.txt` changes.
- throw `ArgumentNullException`, if `owner` is `null`, like we do in `ControlAccessibleObject`:
image

- enable nullablility for `ComboBoxChildEditUiaProvider` and `ComboBoxChildListUiaProvider` as well, if it's possible.
- add a unit test for `ChildAccessibleObject` that the constructor throws ANException, if `owner` is `null`.

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ChildAccessibleObject.cs]: I will do a separate PR when I have time for `ComboBoxChildEditUiaProvider` and `ComboBoxChildListUiaProvider`. For the rest of the comments, I will update the current PR.

Comment (vladimir-krestov) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ChildAccessibleObject.cs]: Thank you!

**Review by RussKie — DISMISSED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ChildAccessibleObject.cs [hunk 3]
```diff
@@ -16,17 +14,17 @@ public class ChildAccessibleObject : AccessibleObject
 
             public ChildAccessibleObject(ComboBox owner, IntPtr handle)
             {
-                Debug.Assert(owner?.IsHandleCreated is true, "ComboBox's handle hasn't been created");
+                _owner = owner.OrThrowIfNull();
 
-                _owner = owner;
+                Debug.Assert(owner.IsHandleCreated is true, "ComboBox's handle hasn't been created");
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ChildAccessibleObject.cs:19]:
```suggestion
                Debug.Assert(owner.IsHandleCreated, "ComboBox's handle hasn't been created");
```

File: src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ChildAccessibleObject.cs [hunk 4]
```diff
@@ -16,17 +14,17 @@ public class ChildAccessibleObject : AccessibleObject
...
 
-                if (owner?.IsHandleCreated is true)
+                if (owner.IsHandleCreated is true)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ChildAccessibleObject.cs:21]:
```suggestion
                if (owner.IsHandleCreated)
```

**Review by RussKie — COMMENTED** [review]
👍 with few nits

**Comment by RussKie** [comment]
Docs: https://github.com/dotnet/docs/issues/26720

**Review by Tanya-Solyanik — APPROVED** [review]

**Review by vladimir-krestov — APPROVED** [review]
Looks good ✔️
